codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/StructureGenerator.kt

```diff
@@ -248,6 +248,11 @@ class StructureGenerator(
                     } else {
                         memberSymbol
                     }
+
+                    if (builderMemberSymbol.shape is BlobShape && builderMemberSymbol.isNotNullable) {
+                        writer.addImport(RuntimeTypes.Core.Text.Encoding.decodeBase64)
+                    }
```

Comment:
decodeBase64 is used in `KotlinSymbolProvider`, but we don't have access to the `KotlinWriter` there, so the import gets added here if necessary. Any other ideas to work around this?

Reply:
Correct, we don't have a `KotlinWriter` (nor should we) in `KotlinSymbolProvider`. But we're still providing symbols and [symbols can declare additional imports](https://github.com/smithy-lang/smithy-kotlin/blob/0f8db4452966822949bbe6616e3642b2063cfb63/codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/SymbolBuilder.kt#L43-L60). We should be setting up those references before returning from the symbol provider.

This probably means a bit of a light refactor in the symbol provider since it's mostly slinging text strings around and wrapping it into a symbol right before returning. But we probably need better awareness of non-textual symbol info in places like `getDefaultValue`.

Reply:
I did a minimal refactor to give `getDefaultValue` access to the Symbol being built, calling `addReferences` where appropriate